File: asignaturas/sql/Asignaturas-IvanRonco/Profesores_Y_Modulos_BD-Java/src/main/java/models/Profesor.java
package models;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@AllArgsConstructor
public class Profesor {
    long id;
    String nombre;
    LocalDateTime fechaIncorporacion;
    List<Modulo> modulos;
}

File: asignaturas/sql/SergioDeIscar-Asignatura-Kotlin/SergioDeIscar-Asignatura-Kotlin/src/main/kotlin/models/Profesor.kt
package models

import java.time.LocalDate

data class Profesor(
    val id: Long = 0,
    val nombre: String,
    val fechaIncorporacion: LocalDate,
    val modulos: List<Modulo>
){
    override fun equals(other: Any?): Boolean {
        if (other == null || other !is Profesor) return false
        return this.id == other.id &&
                this.nombre == other.nombre &&
                this.fechaIncorporacion == other.fechaIncorporacion
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + nombre.hashCode()
        result = 31 * result + fechaIncorporacion.hashCode()
        return result
    }
}